### ISSUE TITLE
Sporadic build failure: ContainerKeyIndex.testRecoveryTimeout timed out

### ISSUE THREAD

**Issue by shiveshr** [pravega/pravega#4203]
**Problem description**
```
io.pravega.segmentstore.server.tables.ContainerKeyIndexTests > testRecoveryTimeout FAILED

    java.util.concurrent.ExecutionException: java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.

        at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)

        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)

        at io.pravega.segmentstore.server.tables.ContainerKeyIndexTests.testRecoveryTimeout(ContainerKeyIndexTests.java:590)

        Caused by:

        java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.
```

**Problem location**
ContainerKeyIndexTests 

**Suggestions for an improvement**
Fix the sporadically failing test

**Comment by shrids** [comment]
Observing this issue again in travis builds

```io.pravega.segmentstore.server.tables.ContainerKeyIndexTests > testRecoveryTimeout FAILED

    java.util.concurrent.ExecutionException: java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.

        at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)

        at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1928)

        at io.pravega.segmentstore.server.tables.ContainerKeyIndexTests.testRecoveryTimeout(ContainerKeyIndexTests.java:590)

        Caused by:

        java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.```

**Comment by RaulGracia** [comment]
Another reproduction in `master` for Git Actions:
```
io.pravega.segmentstore.server.tables.ContainerKeyIndexTests > testRecoveryTimeout FAILED
    java.util.concurrent.ExecutionException: java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.
        at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
        at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022)
        at io.pravega.segmentstore.server.tables.ContainerKeyIndexTests.testRecoveryTimeout(ContainerKeyIndexTests.java:594)

        Caused by:
        java.util.concurrent.TimeoutException: Table Segment 0 recovery timed out.
            at io.pravega.segmentstore.server.tables.ContainerKeyIndex$RecoveryTracker.lambda$setupRecoveryTask$1(ContainerKeyIndex.java:813)
            at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
            at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
            at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
            at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
            at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by andreipaduroiu** [comment]
Included in #4193.
